### DISCUSSION TITLE
dxf2png get white background

### DISCUSSION THREAD

**Discussion by Wananrd** [mozman/ezdxf#1060] [Q&A]
I am a AI student. I am not good at CAD. My project needs to use dxf to png.

I following the document learned how to transform, it succeed in 99.99% .dxf files, but some of them failed.

Please help me find where is problems and how to solve it.

Here is my dxf img and code. 
[address_dxf.zip](https://github.com/mozman/ezdxf/files/14767113/address_dxf.zip)

![image](https://github.com/mozman/ezdxf/assets/70440453/a1165a6f-9fec-4c6b-a800-41351edb437f)

```python
import ezdxf
from ezdxf.addons.drawing import Frontend, RenderContext, pymupdf, layout, config
import os

def dxf2png(file_name, save_folder):
    if file_name.endswith(".dxf"):
        try:
            doc = ezdxf.readfile(filename=file_name)
            # 这里是你的其他处理代码
            msp = doc.modelspace()
            # 1. create the render context
            context = RenderContext(doc)
            # 2. create the backend
            backend = pymupdf.PyMuPdfBackend()
            # 3. create and configure the frontend
            cfg = config.Configuration(background_policy=config.BackgroundPolicy.WHITE)
            frontend = Frontend(context, backend, config=cfg)
            # 4. draw the modelspace
            frontend.draw_layout(msp)
            # 5. create an A4 page layout
            page = layout.Page(891, 891, layout.Units.mm, margins=layout.Margins.all(20))
            # 6. get the PDF rendering as bytes
            png_bytes = backend.get_pixmap_bytes(page, fmt="png", dpi=96)
            # 7. save png
            save_name = file_name.split("\\")[-1].split(".dxf")[0] + ".png"
            save_file = os.path.join(save_folder, save_name)
            with open(save_file, "wb") as fp:
                fp.write(png_bytes)

        except ezdxf.lldxf.const.DXFStructureError as e:
            print("捕获到DXFStructureError异常:", e)
            # 这里是处理异常的代码

if __name__ == "__main__":
    # 单独文件处理
    dxf_file = "3B7P-L12100-1.dxf"
    save_folder = "./result"
    dxf2png(dxf_file, save_folder)`
```

**Comment by mozman** [comment]
This is partly a bug in `ezdxf` v1.2.0 which is already fixed. The `color_policy` is set to `ColorPolicy.BLACK` for this PNG to see yellow on white background.

![3B7P-L12100-1](https://github.com/mozman/ezdxf/assets/4230966/10965565-5c5c-4824-bfc8-05f00976c010)

These four objects are `REGION` entities based on `ACIS` and rendering `ACIS` entities is not supported by `ezdxf`.

![image](https://github.com/mozman/ezdxf/assets/4230966/bbd4e0cc-72e6-4502-a74a-694319cb9320)

**Reply by Wananrd** [reply]
Thanks for your answer and your work.

Sorry my question is unclearly. If I want to show the white line in my .png, you called `REGIN` entities. How could I implement it?

Is there some ways? such as change CAD and edit some module.

**Reply by mozman** [reply]
The `EXPLODE` command in CAD applications converts `REGION` entities into basic DXF entities:

![3B7P-L12100-1](https://github.com/mozman/ezdxf/assets/4230966/7f49f0a7-2e5f-4b57-a73a-0c09c7c9beb5)

**Reply by Wananrd** [reply]
Thank you very much!!!
